Sources/System/Sockets/SocketDescriptor+IPv4Address.swift

```diff
@@ -0,0 +1,139 @@
+/*
+ This source file is part of the Swift System open source project
+
+ Copyright (c) 2021 Apple Inc. and the Swift System project authors
+ Licensed under Apache License v2.0 with Runtime Library Exception
+
+ See https://swift.org/LICENSE.txt for license information
+*/
+
+extension SocketDescriptor {
+  @frozen
+  public struct IPv4Address: RawRepresentable {
+    public var rawValue: CInterop.SockAddrIn
+
+    public init(rawValue: CInterop.SockAddrIn) {
+      self.rawValue = rawValue
+    }
+
+    public init?(_ address: SocketDescriptor.Address) {
+      guard address.domain == Domain.ipv4 else { return nil }
+      let value: CInterop.SockAddrIn? = address.withUnsafeBytes { buffer in
+        guard buffer.count >= MemoryLayout<CInterop.SockAddrIn>.size else {
+          return nil
```

Comment:
When might this happen?

Reply:
It is possible to manually initialize a SocketAddress with garbage data.

Reply:
Arguably we should reject overlong addresses, but evidently people overallocate sometimes.